src/main/java/datawave/microservice/query/storage/TaskStates.java

```diff
@@ -0,0 +1,284 @@
+package datawave.microservice.query.storage;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.zaxxer.sparsebits.SparseBitSet;
```

Comment:
Do we need the memory optimization that using a sparsebitset provides over a hashset or other structure? I do see the memory benefits, but will the memory optimization be so great that it accounts for this class being harder to grok for someone not familiar?

Reply:
this worries me as well

Reply:
This was also for performance, and to minimize storage.  However I am happy to attempt to replace with something else.

Reply:
Replacing with a sorted set of integers...

Reply:
done